test/test_pipeline_yaml.py

```diff
@@ -792,3 +847,45 @@ def test_save_yaml_overwrite(tmp_path):
     with open(tmp_path / "saved_pipeline.yml", "r") as saved_yaml:
         content = saved_yaml.read()
         assert content != ""
+
+
+def test_save_yaml_with_custom_node_kwargs(tmp_path):
+    class CustomNode(MockNode):
+        def __init__(self, param: int):
+            super().__init__()
+            self.param = param
+
+    pipeline = Pipeline()
+    pipeline.add_node(CustomNode(param=1), name="custom_node", inputs=["Query"])
+    pipeline.save_to_yaml(tmp_path / "saved_pipeline.yml")
+
+    print(pipeline.get_node("custom_node")._component_config)
```

Comment:
Is this part of the test or just for debugging purpose?

Reply:
Definitely debugging, I'm removing it :smile: